radsat-sk/operation/subsystems/RBattery.c

```diff
@@ -0,0 +1,279 @@
+
+	// Create a temporary array for calculated temperatures before transferring into the BatteryStatus structure
+	float storedTempData[NUMBER_OF_TEMP_COMMANDS] = {0};
+
+
+	// Send 3 commands to get ADC output voltage readings from the Battery
+	for(int i = 0; i < NUMBER_OF_VOLTAGE_AND_CURRENT_COMMANDS; i = i + 1){
+		memcpy(i2c_command, &batteryVoltageCommandBytes[i], BATTERY_TELEM_COMMAND_LENGTH);
+		int error = batteryTalk(i2c_command, i2c_data);
+
+		if(error != 0){
+			// TODO: Some Error Raising here?
+			// return the error
+			return error;
+		}
+
+		memcpy(&storedData[i], i2c_data, BATTERY_RESPONSE_LENGTH);
+	}
+	// Get ADC Output Voltages
+	dataStorage->outputVoltageBatteryBus = ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE * storedData[0];
+	dataStorage->outputVoltage5VBus = ADC_COUNT_TO_5V_BUS_OUTPUT_VOLTAGE * storedData[1];
+	dataStorage->outputVoltage3V3Bus = ADC_COUNT_TO_3V3_BUS_OUTPUT_VOLTAGE * storedData[2];
+
+
+	// Send 3 commands to get ADC output current readings from the Battery
+	for(int i = 0; i < NUMBER_OF_VOLTAGE_AND_CURRENT_COMMANDS; i = i + 1){
+		memcpy(i2c_command, &batteryCurrentCommandBytes[i], BATTERY_TELEM_COMMAND_LENGTH);
+		int error = batteryTalk(i2c_command, i2c_data);
+
+		if(error != 0){
+			// TODO: Some Error Raising here?
+			// return the error
+			return error;
+		}
+
+		memcpy(&storedData[i], i2c_data, BATTERY_RESPONSE_LENGTH);
+	}
+	// Get ADC Output Currents
+	dataStorage->outputCurrentBatteryBus = ADC_COUNT_TO_BATTERY_OUTPUT_CURRENT_MAG * storedData[0];
+	dataStorage->outputCurrentBatteryBus = ADC_COUNT_TO_5V_BUS_CURRENT_DRAW * storedData[1];
+	dataStorage->outputCurrentBatteryBus = ADC_COUNT_TO_3V3_BUS_CURRENT_DRAW * storedData[2];
+
+
+	// Send 4 commands to get ADC output current readings from the Battery
+	for(int i = 0; i < NUMBER_OF_TEMP_COMMANDS; i = i + 1){
+		memcpy(i2c_command, &batteryTemperatureCommandBytes[i], BATTERY_TELEM_COMMAND_LENGTH);
+		int error = batteryTalk(i2c_command, i2c_data);
+
+		if(error != 0){
+			// TODO: Some Error Raising here?
+			// return the error
+			return error;
+		}
+
+		memcpy(&storedTempData[i], i2c_data, BATTERY_RESPONSE_LENGTH);
+	}
+	// Get ADC Temperatures
+	dataStorage->motherboardTemp = (ADC_COUNT_TO_MOTHERBOARD_TEMP_SCALE * storedTempData[0])
+			- ADC_COUNT_TO_MOTHERBOARD_TEMP_SHIFT;
+
+	dataStorage->daughterboardTemp1 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * storedTempData[1])
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	dataStorage->daughterboardTemp2 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * storedTempData[2])
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	dataStorage->daughterboardTemp3 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * storedTempData[3])
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	return 0;
+}
+
+
+/**
+ * Send a telecommand via I2C to check on the current battery voltage, and if under 6.5V, raise the global safeFlag
+ * @return nothing.
+ */
+int checkSafeFlag(void){
```

Comment:
Style guide: change to `batteryIsSafe` or `batteryIsNotSafe` (whichever is more accurate) or similar

Reply:
Also, should this function offer a way for the calling API to know whether the flag was raised? Who is the inteded caller of this function?

Reply:
Also pt2: Update the docstring as it is outdated (claims no return value)